portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,18 +15,30 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import com.google.gson.Gson;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Anooj!</h1>");
+    ArrayList<String> messages = new ArrayList<String>();
```

Comment:
This is a super Java nit...only make the change if you want to.

It's preferable to declare variables of types that represent the *interface* of the data structure you want to use instead of a concrete implementation. So in this case, it would often be written as:

```
List<String> messages = new ArrayList<String>();
...
```

and then

```
private String convertToJson(List<String> list) {...}
```

Reply:
Sorry I didn't fully understand this. I read this [article](https://techdifferences.com/difference-between-list-and-arraylist-in-java.html#:~:text=).  I learned that List is an Interface and the Class is ArrayList. Is it better to always declare variables as an interface? When (if at all) would it be appropriate to declare using the class. Or is it just a bad practice?

Reply:
Based on our discussion on chat, resolving this comment for now; we'll discuss this issue later.